### PR TITLE
Add some basic Windows builds to the Windows CI workflow

### PR DESCRIPTION
**Pull request by t8m**
We need them to catch regressions like #15347

### CODE REVIEW

**Comment by t8m** [comment]
>  Assuming the CIs pass....

They supposedly won't until #15347 is fixed. But I won't merge before that.

**Comment by t8m** [comment]
@paulidale I had to remove the --strict-warnings on the x86 build as that is broken due to size_t -> int conversion in apps/lib/apps.c

**Comment by paulidale** [comment]
Fixing such conversions would be better.  For another PR though.

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Hmm, unfortunately there is some further breakage in the fips builds. Not sure why this does not appear on AppVeyor.

**Comment by t8m** [comment]
@paulidale Tests seem to be fine now.

Unfortunately I had to disable fips due to some error with creation of the fipsmodule.cnf - not sure why it works fine on AppVeyor.

Also the tests take quite some time to complete, which might be a problem if we want a quick turnaround on the CI for PRs.

**Comment by t8m** [comment]
In particular the shared-x86 test took 40 minutes.

**Comment by DDvO** [comment]
> In particular the shared-x86 test took 40 minutes.

I still have a solution in the pipeline to parallelize the compilations invoked by `nmake` within `cl`.

**Comment by t8m** [comment]
> I still have a solution in the pipeline to parallelize the compilations invoked by `nmake` within `cl`.

This won't help much as the main part of the time are the tests.

**Comment by DDvO** [comment]
> > I still have a solution in the pipeline to parallelize the compilations invoked by `nmake` within `cl`.
> 
> This won't help much as the main part of the time are the tests.

Oh!

Another thing is avoiding needless non-EC key generation as started in #13715.

**Review by slontis — COMMENTED** [review]
It may be worth adding a link to
https://github.com/ilammy/msvc-dev-cmd

Have a look at the arch settings..

**Comment by slontis** [comment]
> Unfortunately I had to disable fips due to some error with creation of the fipsmodule.cnf - not sure why it works fine on AppVeyor.

I wonder if it related to the ````working_directory: _build````` line?

**Comment by t8m** [comment]
> I wonder if it related to the ''working_directory: _build' line?

Hmm, I could try in-tree build if it makes any difference.

**Comment by t8m** [comment]
Unfortunately the in-tree build did not help with the FIPS issue.

**Comment by t8m** [comment]
@paulidale please re-review

**Review by paulidale — APPROVED** [review]

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -3,6 +3,54 @@ name: Windows GitHub CI
...
+    - uses: actions/checkout@v2
+    - uses: ilammy/msvc-dev-cmd@v1
+      with:
+        arch: ${{ matrix.arch }}
+    - uses: ilammy/setup-nasm@v1
+      with:
+        platform: ${{ matrix.arch }}
+    - name: prepare the build directory
+      run: mkdir _build
+    - name: config
```
Comment (levitte) [inline .github/workflows/windows.yml:24]: How about this?

``` yaml
    - name: config for ${{ matrix.arch }}
```

Comment (t8m) [reply .github/workflows/windows.yml:24]: Why, you can see the matrix.arch in the build name anyway: https://github.com/openssl/openssl/pull/15349/checks?check_run_id=2638029715

**Comment by t8m** [comment]
Merged to master. Thank you for the reviews!

**Pull request closed without merging**